src/shapes.ts

```diff
@@ -0,0 +1,361 @@
+/**
+ * @packageDocumentation
+ * @module settings
+ *
+ * This file consists of shapes supported by chemiscope via 3dmol
+ *
+ * To add a new shape type, extend the `Shape` class and define the
+ * inherited functions (`validateParams` and `outputTo3Dmol`).
+ * `validateParams` should check a `ShapeData` object to see that
+ * it has the correct parameters for the given type. `outputTo3Dmol`
+ * should take a color and output a `CustomShapeSpec` (https://3dmol.csb.pitt.edu/doc/types.html#CustomShapeSpec)
+ * object. You will also need to add any parameters for your shape
+ * to ShapeData.
+ *
+ * To add a new shape, you will also need to add code to `structure/viewer.ts`.
+ */
+
+import { default as $3Dmol } from '3dmol';
+import { Quaternion } from '3dmol';
+
+// Just an interface to enforce XYZ-type coordinates
+export interface XYZ {
+    x: number;
+    y: number;
+    z: number;
+}
+export interface SphereData {
+    radius: number;
+    kind: string;
+}
+
+export interface EllipsoidData {
+    semiaxes: [number, number, number];
+    orientation?: [number, number, number, number];
+    kind: string;
+}
+
+export interface PolyData {
+    vertices: [number, number, number][];
+    simplices: [number, number, number][];
+    orientation?: [number, number, number, number];
+    kind: string;
+}
+
+export type ShapeData = SphereData | EllipsoidData | PolyData;
+
+export class Shape {
+    /// orientation of the particle
+    public position: [number, number, number];
+    public quaternion: Quaternion;
+
+    constructor(
+        pos: [number, number, number],
+        orientation: [number, number, number, number] = [1, 0, 0, 0]
+    ) {
+        this.quaternion = new Quaternion(
+            orientation[1],
+            orientation[2],
+            orientation[3],
+            orientation[0]
+        );
```

Comment:
Why not define our quaternions in the same order as 3dmol? i.e.
```suggestion
        this.quaternion = new Quaternion(
            orientation[0]
            orientation[1],
            orientation[2],
            orientation[3]
        );
```

Reply:
Will document the convention.

In the field, there's a lot of disagreement between wxyz and xyzw -- I'm from the camp of xyzw, as that makes things simplest with codes I use.

Reply:
Added some notes!

Reply:
You mean codes you use are based on `wxyz`, right? it feels to me we should rather follow the OpenGL convention where vec4 are `xyzw` (https://www.khronos.org/opengl/wiki/Data_Type_(GLSL)#Vectors). We can make sure everything works fine by converting everything on the Python side, so end users do not have to worry about it.

Reply:
So you mean having quaternion changers in the chemiscope interpreter?

Reply:
Okay, so I've implemented xyzw across the board and documented as such!